cfaster/scrapers/utils/urls.py

```diff
@@ -3,15 +3,48 @@
 
 logger = logging.getLogger(__name__)
 
+def url_filter(string, to_remove):
+    if string.startswith(to_remove):
+        return string[len(to_remove):]
+    else:
+        return string
+
 
 def validate_url(url, expected_domain):
     logger.info('Validating url {} with expected domain {}'.format(
         url, expected_domain))
     url = url.strip()
-    reg = r'^(?:https?://)?(?:www.)?(\w+)(?:.\w+)+(?:/\w*)*$'
-    match = re.search(reg, url)
-    if match is None:
+
+    if expected_domain not in url:
         return False
-    if len(match.groups()) and match.group(1) == expected_domain:
-        return True
-    return False
+
+    for leader in ('https://', 'http://', 'www', 'ftp://'):
+        url = url_filter(url, leader)
+
+    # Split off everything after the last domain
+    url = url.split('/')[0]
+
+    # ... and split up the domains
+    url = url.split('.')
+
+    if expected_domain not in url:
+        return False
+
+    for loc, url_piece in enumerate(url[::-1]):
+        if loc == 0:
+            continue
+        elif loc == 1:
+            if url_piece == expected_domain:
+                return True
+            if len(url_piece) > 3:
+                return False
+
+        elif loc == 2:
+            if url_piece == expected_domain:
+                return True
+            if len(url_piece) > 3:
+                return False
+
+        else:
+            return False
```

Comment:
The `if loc == 0:` is not a good practice, I see the intention of ignoring the top level domains (`.com`, `.net`, etc) but if this was going to be really needed would be a smarter choice to iterate through `list[1:]` instead since you always ignore the first.
An even better approach is to remove all top level domains which could be given from the string: `top_level_domains = ['.com', '.net', '.edu', ... ]`. It would be done the same way we remove the `leaders`.

Reply:
Should I instead add in just the [list of possible TLDs](http://data.iana.org/TLD/tlds-alpha-by-domain.txt) since there could be quite a few?

Reply:
Good idea but it looks like we could get a lot of false positives. Maybe since we are using only [codeforces](codeforces.com) as a scrape target we could only consider `.com` TLD.

Reply:
Oh!  Ok, haha that'll greatly simplify things :)